app/controllers/groups_controller.rb

```diff
@@ -420,12 +420,19 @@ def remove_member
   end
 
   def request_membership
-    params.require(:reason)
+    params.require(:reason) if !params[:topic_id].present?
 
     group = find_group(:id)
 
+    if params[:topic_id].present?
+      topic_url = Topic.find(params[:topic_id].to_i).url
```

Comment:
This will raise an exception if that topic_id does not exist won't it?

Reply:
[1] pry(main)> {}[:topic_id].present?
=> false

Reply:
I guess I would write this way:

```ruby
if params[:topic_id] && topic = Topic.find(params[:topic_id].to_i)
  reason = I18n.t("groups.view_hidden_topic_request_reason", topic_url: topic.url)
end
```

Reply:
Yes the `present?` check is fine, I was more worried if the id was `10123012031203` and no topic could be returned.